OpenRA.Mods.Common/Traits/Player/BaseAttackNotifier.cs

```diff
@@ -60,12 +60,22 @@ public void Damaged(Actor self, AttackInfo e)
 			if (e.Attacker == self.World.WorldActor)
 				return;
 
+			// Only track last hit against our base
+			if (!self.Info.HasTraitInfo<BuildingInfo>())
+				return;
+
 			if (e.Attacker.Owner.IsAlliedWith(self.Owner) && e.Damage <= 0)
 				return;
 
 			if (self.World.WorldTick - lastAttackTime > info.NotifyInterval * 25)
 			{
-				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech", info.Notification, self.Owner.Faction.InternalName);
+				var rules = self.World.Map.Rules;
+				Game.Sound.PlayNotification(rules, self.Owner, "Speech", info.Notification, self.Owner.Faction.InternalName);
+
+				if (info.AllyNotification != null)
+					foreach (Player p in self.World.Players)
+						if (p != self.Owner && p.IsAlliedWith(self.Owner))
```

Comment:
Do we want to play the notification to the player that is attacking, too (in case of friendly fire, which might either be deliberate or not)?

If yes, then :+1:, otherwise you'd need to compare `p` with `e.Attacker.Owner` here as well.

I haven't quite formed an opinion yet on what I'd prefer.